cdap-cli/src/main/java/co/cask/cdap/cli/command/app/ListAppVersionsCommand.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli.command.app;
+
+import co.cask.cdap.cli.ArgumentName;
+import co.cask.cdap.cli.CLIConfig;
+import co.cask.cdap.cli.ElementType;
+import co.cask.cdap.cli.util.AbstractAuthCommand;
+import co.cask.cdap.cli.util.RowMaker;
+import co.cask.cdap.cli.util.table.Table;
+import co.cask.cdap.client.ApplicationClient;
+import co.cask.common.cli.Arguments;
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+
+import java.io.PrintStream;
+import java.util.List;
+
+/**
+ * Lists all versions of a specific app.
+ */
+public class ListAppVersionsCommand extends AbstractAuthCommand {
+
+  private final ApplicationClient appClient;
+
+  @Inject
+  public ListAppVersionsCommand(ApplicationClient appClient, CLIConfig cliConfig) {
+    super(cliConfig);
+    this.appClient = appClient;
+  }
+
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    String appName = arguments.get(ArgumentName.APP.toString());
+    List<String> versions = appClient.listAppVersions(cliConfig.getCurrentNamespace(), appName);
+
+    Table table = Table.builder()
+      .setHeader("version")
+      .setRows(versions, new RowMaker<String>() {
+        @Override
+        public List<?> makeRow(String object) {
```

Comment:
Which is parameterized? The input is just String of version, should have renamed it as version.